assembl/static2/js/app/components/common/socialShare.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { ShareButtons, generateShareIcon } from 'react-share';
+import { I18n } from 'react-redux-i18n';
+
+const {
+  FacebookShareButton,
+  GooglePlusShareButton,
+  LinkedinShareButton,
+  TwitterShareButton,
+  EmailShareButton,
+  WhatsappShareButton,
+  TelegramShareButton
+} = ShareButtons;
+const FacebookIcon = generateShareIcon('facebook');
+const GooglePlusIcon = generateShareIcon('google');
+const LinkedinIcon = generateShareIcon('linkedin');
+const TwitterIcon = generateShareIcon('twitter');
+const EmailIcon = generateShareIcon('email');
+const WhatsappIcon = generateShareIcon('whatsapp');
+const TelegramIcon = generateShareIcon('telegram');
+
+export default class SocialShare extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      copied: false
+    };
+  }
+  render() {
+    const { url, onClose, social } = this.props;
+    return (
+      <div className="share-buttons-container">
+        {social
+          ? <div className="social-share-buttons-container">
+            <div className="social-share-button">
+              <EmailShareButton url={url} onShareWindowClose={onClose}>
```

Comment:
Seeing as to how you have to pass the same props everywhere, let's make a Higher Order Component (https://reactjs.org/docs/higher-order-components.html) that abstracts away the passing of these props.
```
const superShareButton = (Component, data) => {
  return (props) => {
    <Component {...data} {...props} />
  }
}
```

And use it with this example:

```
const EmailShare = superShareButton(EmailShareButton, {url: url, onShareWindowClose: onClose, size: 32});
return (<EmailShare />);
```

Reply:
Thought, right now, I'm realizing this doesn't reduce your code. Well, take it as a learning opportunity for HOCs.

Reply:
Okay, I was too tired to give a good example of HOC. Here is the simplified solution:
```
const superShareButton = (Component) => {
  const url = 'someUrl';
  const size = 32;
  const onClose = () => console.log('something magical happened');
  const data = {url: url, size: size, onShareWindowClose: onClose}
  return (props) => {
    <Component {...data} {...props} />
  }
}
```

Then you can simply write:

```
const EmailShare = superShareButton(EmailShareButton);
return (<EmailShare />);
...
```